idseq_dag/util/s3.py

```diff
@@ -119,39 +115,39 @@ def fetch_from_s3(src,
                         allow_s3mi = False
 
                 if untar:
-                    command_params = " | tar xvf - -C {output_dir}".format(output_dir=destdir)
+                    command_params = f" | tar xvf - -C {destdir}"
                 else:
                     pipe_filter = ""
                     if unzip:
                         pipe_filter = "| gzip -dc "
-                    command_params = "{pipe_filter} > {destination}".format(
-                        pipe_filter=pipe_filter, destination=dst)
-                log.write("command_params: %s" % command_params)
+                    command_params = f"{pipe_filter} > {dst}"
+                log.write(f"command_params: {command_params}")
 
                 try:
                     assert allow_s3mi
-                    cmd = "s3mi cat {source} {command_params}".format(
-                        source=src, command_params=command_params)
+                    cmd = f"s3mi cat {src} {command_params}"
                     command.execute(cmd)
                 except:
                     log.write(
                         "Failed to download with s3mi. Trying with aws s3 cp..."
                     )
-                    cmd = "aws s3 cp --quiet {source} - {command_params}".format(
-                        source=src, command_params=command_params)
+                    cmd = f"aws s3 cp --quiet {src} - {command_params}"
                     command.execute(cmd)
                 return dst
             except subprocess.CalledProcessError:
                 # Most likely the file doesn't exist in S3.
                 log.write(
                     "Failed to fetch file from S3. Most likely does not exist."
                 )
+                # Delete potential empty file remnant
+                if os.path.isfile(dst) and os.stat(dst).st_size == 0:
+                    os.remove(dst)
```

Comment:
This is cool.   I now think we should just "rm -rf {dst}".   And, caveat emptor, if your tar file is named "foo.tar" it better unpack under "foo/..." or else...  but that's already an existing caveat & warning in the code.

Reply:
You can use `command.remove_file` (or `command.remove_rf` as Boris suggested), because they will also log the execution.